osu.Game/Screens/Play/BreakOverlay.cs

```diff
@@ -109,6 +116,13 @@ protected override void LoadComplete()
             initializeBreaks();
         }
 
+        protected override void Update()
+        {
+            base.Update();
+
+            IsBreakTime.Value = breaks.Where(b => b.HasEffect).Any(b => Clock.CurrentTime >= b.StartTime && Clock.CurrentTime <= b.EndTime);
```

Comment:
this can't exist. linq can't be used in hot paths like this – it will be allocating memory and performing badly.

you will have to keep a running track of the current break index and make this an efficient lookup.